src/components/ContentNode/SectionTitle.vue

```diff
@@ -0,0 +1,75 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <component
+   :id="anchor ? anchor : null"
+   :is="tag"
+   class="section-title"
+  >
+    <a
+      v-if="anchor || href"
+      :href="`#${anchor || href}`"
```

Comment:
I would suggest you look at how we have done this in the past, like in the TutorialsNavigationLink component.

We use a `scrollToElement` function that actually adjusts the scrolling by a few pixels, so the navigation does not overlap the title. I saw you added `scroll-margin-top` but I am getting cutoffs on the Topic section

We also used to use `<router-link>` components instead of `a`, not sure if there is a major diff.

![image](https://user-images.githubusercontent.com/9863944/184084450-a4774561-20a3-4c8e-9a64-2bcebee839e9.png)

Reply:
So if we pass `anchor`, it adds an ID, and if we pass just `href` it does not, and it uses both to create a `href`?

I had to go back midway review when I finally got whats going on, felt confusing. I will leave it the the rest of the team to give more feedback.

Reply:
I addressed the feedback and I fixed the confusion with the anchor and href prop. Now we only have one. Check it out again for another review, thanks!